src/Http/Controllers/RestApiController.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace DoubleThreeDigital\Runway\Http\Controllers;
+
+use DoubleThreeDigital\Runway\Http\Resources\ApiResource;
+use DoubleThreeDigital\Runway\Resource;
+use DoubleThreeDigital\Runway\Runway;
+use Facades\Statamic\API\FilterAuthorizer;
+use Illuminate\Support\Str;
+use Statamic\Exceptions\NotFoundHttpException;
+use Statamic\Http\Controllers\API\ApiController;
+
+class RestApiController extends ApiController
+{
+    private $config;
+
+    protected $resourceConfigKey = 'runway';
+
+    protected $routeResourceKey = 'handle';
+
+    protected $resourceHandle;
+
+    public function index($handle)
+    {
+        $this->abortIfDisabled();
+
+        $this->resourceHandle = Str::singular($handle);
+
+        $resource = Runway::findResource($this->resourceHandle);
+
+        if (! $resource) {
+            throw new NotFoundHttpException;
+        }
+
+        $results = $this->filterSortAndPaginate($resource->model()->query());
+        $results->setCollection($results->getCollection()->map(fn ($model) => $this->makeResourceFromModel($resource, $model)));
+
+        return ApiResource::collection($results);
+    }
+
+    public function show($handle, $id)
+    {
+        $this->abortIfDisabled();
+
+        $this->resourceHandle = Str::singular($handle);
+
+        $resource = Runway::findResource($this->resourceHandle);
+
+        if (! $resource) {
+            throw new NotFoundHttpException;
+        }
+
+        if (! $model = $resource->model()->find($id)) {
+            throw new NotFoundHttpException;
+        }
+
+        return ApiResource::make($this->makeResourceFromModel($resource, $model));
+    }
+
+    protected function allowedFilters()
+    {
+        return FilterAuthorizer::allowedForSubResources('api', $this->resourceConfigKey, Str::plural($this->resourceHandle));
+    }
+
+    private function makeResourceFromModel($resource, $model)
+    {
+        if (! $this->config) {
+            $this->config = collect(config('runway.resources'))->get(get_class($model));
+        }
+
+        return new Resource(
```

Comment:
I'm a little confused as to why we're new-ing up a resource here, rather than using the ones discovered by the `Runway` class.

Reply:
Mainly just to be able to use the query builder stuff. Happy to do it differently if there's a better way.

Reply:
Would [`Runway::findResourceByModel`](https://github.com/duncanmcclean/runway/blob/6.x/src/Runway.php#L84) do what you need?